src/cats/labs/channel.cljc

```diff
@@ -78,64 +64,47 @@
 
     p/Functor
     (-fmap [_ f mv]
-      (let [c (a/chan 1 (map f))]
-        (a/pipe mv c)
-        c))
+      (a/pipe mv (a/chan 1 (map f))))
 
     p/Semigroup
     (-mappend [_ sv sv']
-      (chain-chans sv sv'))
+      (a/merge [sv sv']))
```

Comment:
I know that and I understand the advantages of using merge. My unique concern is about correctness vs practical approach. How is correct you approach in terms of mappend laws?